lib/csgo_stats/logs/parser.ex

```diff
@@ -105,13 +106,168 @@ defmodule CsgoStats.Logs.Parser do
     end
   end
 
+  # Example: "casual"
+  game_mode =
+    choice([
+      string("casual"),
+      string("competitive")
+    ])
+
+  # Example: "de_inferno"
+  game_map =
```

Comment:
Is there a reason why this isn't just searching for anything except `"`?

Reply:
Yeah, this is basically a choice between two evils: accepting any input or only accepting expected input. 

I've chosen here to be very strict about what we accept as valid input.

Either we're (like I've chosen here). This has an obvious con: we need a code change to add support for new maps. But there's also a big pro: downstream code will only have to handle a known set of cases. For example, if we add background pictures for maps, we can look at the parser and know exactly which maps we need a picture of.

In fact, I'm considering if we should translate all such "constant strings" into atoms. And maybe extract the definition of accepted maps into a Map module. Then we could have a module for all such game entities (Player, Map, Weapon) which define a struct with fields and/or list of valid atoms...

Reply:
If we have a "fallback" map with a default picture or such then I don't see any pros for handling them specifically. 
There could be 1v1 tournaments that are played on maps outside the usual map pool. I just think it would be a shame not being able to handle these maps.

I like the module idea 👍